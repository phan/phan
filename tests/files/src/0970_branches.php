<?php

function unused() {
	$var = 1;
}

function overwritten_no_branch() {
	$var = 1;
	$var = 2;
	return $var;
}

function overwritten_if() {
	$var = 1;
	if (rand(0,1)) {
		$var = 2;
	} else {
		$var = 3;
	}
	return $var;
}

function overwritten_switch() {
	$var = 1;
	switch(rand(0,1)) {
		case 0:
			$var = 2;
			break;

		case 1:
			$var = 3;
			break;

		default:
			$var = 4;
			break;
	}
	return $var;
}

function overwritten_conditional() {
	$var = 1;
	(rand(0,1) == 1) ? $var = 2 : $var = 3;
	return $var;
}

function overwritten_try_catch() {
	$var = 1;
	try {
		$var = 2;
	} catch (Exception $_) {
		return;
	}
	return $var;
}