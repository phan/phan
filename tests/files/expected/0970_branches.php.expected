%s:4 PhanUnusedVariable Unused definition of variable $var
%s:8 PhanUnusedVariable Unused definition of variable $var
%s:14 PhanUnusedVariableOverwrittenAllBranches Definition of variable $var is overwritten on all branches
%s:24 PhanUnusedVariableOverwrittenAllBranches Definition of variable $var is overwritten on all branches
%s:42 PhanUnusedVariableOverwrittenAllBranches Definition of variable $var is overwritten on all branches
%s:48 PhanUnusedVariableOverwrittenAllBranches Definition of variable $var is overwritten on all branches
